drivers/include/can/candev.h

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (C) 2016 OTA keys S.A.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for
+ * more details.
+ */
+
+/**
+ * @defgroup    drivers_can
+ * @{
+ *
+ * This interface is supposed to be a low-level interface for CAN drivers.
+ *
+ * @file
+ * @brief       Definitions low-level CAN driver interface
+ *
+ * @author      Vincent Dupont <vincent@otakeys.com>
+ * @author      Toon Stegen <toon.stegen@altran.com>
+ */
+
+#ifndef CANDEV_H
+#define CANDEV_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "can/pkt.h"
+#include "can/common.h"
+
+
+/**
+ * @brief   Possible event types that are send from the device driver to the
+ *          upper layer
+ */
+typedef enum {
+    CANDEV_EVENT_NOEVENT,
+    CANDEV_EVENT_ISR,              /**< driver needs it's ISR handled */
+    CANDEV_EVENT_WAKE_UP,          /**< driver has been woken up by bus */
+    CANDEV_EVENT_TX_CONFIRMATION,  /**< a packet has been sent */
+    CANDEV_EVENT_TIMEOUT_TX_CONF,  /**< tx conf timeout received */
+    CANDEV_EVENT_RX_INDICATION,    /**< a packet has been received */
+    CANDEV_EVENT_TX_ERROR,         /**< there was an error when transmitting */
+    CANDEV_EVENT_RX_ERROR,         /**< there was an error when receiving */
+    CANDEV_EVENT_BUS_OFF,          /**< bus-off detected */
+    CANDEV_EVENT_ERROR_PASSIVE,    /**< driver switched in error passive */
+    CANDEV_EVENT_ERROR_WARNING,    /**< driver reached error warning */
+    /* expand this list if needed */
+} candev_event_t;
+
+/**
+ * @brief   Forward declaration for candev struct
+ */
+typedef struct candev candev_t;
+
+/**
+ * @brief   Event callback for signaling event to upper layers
+ *
+ * @param[in] type          type of the event
+ * @param[in] arg           event argument
+ */
+typedef void (*candev_event_cb_t)(candev_t *dev, candev_event_t event, void *arg);
+
+/**
+ * @brief Structure to hold driver state
+ *
+ * Supposed to be extended by driver implementations.
+ * The extended structure should contain all variable driver state.
+ */
+struct candev {
+    const struct candev_driver *driver;    /**< ptr to that driver's interface. */
+    candev_event_cb_t event_callback;      /**< callback for device events */
+    void *isr_arg;                         /**< argument to pass on isr event */
+    kernel_pid_t pid;                      /**< device thread pid */
```

Comment:
is this necessary?

netdev2 has been designed to not force usage of a thread per device. maybe that architecture can be re-used here?